src/sql/workbench/browser/modal/calloutDialog.ts

```diff
@@ -0,0 +1,333 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { URI } from 'vs/base/common/uri';
+import * as TelemetryKeys from 'sql/platform/telemetry/common/telemetryKeys';
+import * as DOM from 'vs/base/browser/dom';
+import * as styler from 'vs/platform/theme/common/styler';
+import * as strings from 'vs/base/common/strings';
+import { attachButtonStyler } from 'sql/platform/theme/common/styler';
+import { IDialogProperties, Modal } from 'sql/workbench/browser/modal/modal';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { localize } from 'vs/nls';
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { IClipboardService } from 'vs/platform/clipboard/common/clipboardService';
+import { IFileDialogService, IOpenDialogOptions } from 'vs/platform/dialogs/common/dialogs';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { ILogService } from 'vs/platform/log/common/log';
+import { ITextResourcePropertiesService } from 'vs/editor/common/services/textResourceConfigurationService';
+import { IAdsTelemetryService } from 'sql/platform/telemetry/common/telemetry';
+import { attachModalDialogStyler } from 'sql/workbench/common/styler';
+import { ILayoutService } from 'vs/platform/layout/browser/layoutService';
+import { Deferred } from 'sql/base/common/promise';
+import { InputBox } from 'sql/base/browser/ui/inputBox/inputBox';
+import { Button } from 'sql/base/browser/ui/button/button';
+import { Checkbox } from 'sql/base/browser/ui/checkbox/checkbox';
+import { RadioButton } from 'sql/base/browser/ui/radioButton/radioButton';
+import { IPathService } from 'vs/workbench/services/path/common/pathService';
+
+export type CalloutType = 'IMAGE' | 'LINK';
```

Comment:
based on the file location, this should be a generic callout dialog, but these are notebook specific logic, this should be moved to notebook related folders. after this still needs to be refactored so that IMAGE and LINK are not aware of each other, you can have imageCalloutDialog and LinkCalloutDialog, but the base class shouldn't know these.

Reply:
@Charles-Gagnon and I discussed this. I will create a follow up PR to make changes like this. Both modal and calloutDialog need to be revised to work as base classes and then specific dialogs can be new superclasses - extensions of these base classes.